### PR TITLE
Implement Error for DecodeError and BufReadDecoderError.

### PR DESCRIPTION
**Pull request by dstu**
This is a basic impl of Error (and Display). It should address #4.

### CODE REVIEW

File: src/read.rs [hunk 1]
```diff
@@ -30,6 +33,24 @@ impl<'a> BufReadDecoderError<'a> {
     }
 }
 
+impl<'a> fmt::Display for BufReadDecoderError<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+      match *self {
+          BufReadDecoderError::InvalidByteSequence(_) => write!(f, "invalid byte sequence"),
```
Comment (wfraser) [inline src/read.rs]: consider also including what the bad bytes are:
```suggestion
          BufReadDecoderError::InvalidByteSequence(bytes) => write!(f, "invalid byte sequence {:x?}", bytes),
```

Comment (dstu) [reply src/read.rs]: Good call. I also added some padding and reformatted a tiny bit, to respect rustfmt's default line length preferences.

File: src/lib.rs [hunk 2]
```diff
@@ -28,6 +30,25 @@ pub enum DecodeError<'a> {
     },
 }
 
+impl<'a> fmt::Display for DecodeError<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            DecodeError::Invalid { valid_prefix, invalid_sequence, remaining_input, } => {
+                write!(f, "invalid byte sequence (preceding valid string of length {}: '{}'; \
```
Comment (SimonSapin) [inline src/lib.rs]: I parse this as "invalid sequence that precedes a valid string, which is backward. Could you rephrase this? Also, maybe reduce repetition. Something like: `found invalid byte sequence {:02x?} after {} valid bytes, followed by {} more unprocessed bytes`

(Similarly below for incomplete)

**Comment by SimonSapin** [comment]
Hi! Thanks for the PR. If you’d like this change on crates.io, please also increment the version number in `Cargo.toml`.

File: src/lib.rs (hunk 2 above)
Comment (dstu) [reply src/lib.rs]: Done. I also hit it with `cargo fmt` again (which is the reason for the syntactic churn).

**Comment by dstu** [comment]
Happy to contribute. Thanks for the review.

**Comment by SimonSapin** [comment]
Thanks!
